File: app/src/main/java/org/deiverbum/app/data/entity/LHResponsoryEntity.java
package org.deiverbum.app.data.entity;

import static org.deiverbum.app.utils.Constants.LH_RESPONSORY;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import org.deiverbum.app.model.LHResponsory;
import org.deiverbum.app.utils.Utils;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */

@Entity(tableName = LH_RESPONSORY/*,
        indices={
                @Index(value={"texto","fuente", "tipo"},unique = true)}*/
)
public class LHResponsoryEntity {
    @NonNull
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "responsoryID")
    public Integer responsorioId = 0;
    @NonNull
    @ColumnInfo(name = "text")
    public String texto = "";
    @NonNull
    @ColumnInfo(name = "source")
    public String fuente = "";
    @NonNull
    @ColumnInfo(name = "type")
    public Integer tipo = 0;

    @NonNull
    public String getTexto() {
        return texto;
    }

    @NonNull
    public String getFuente() {
        return fuente;
    }

    @NonNull
    public Integer getTipo() {
        return tipo;
    }

    public LHResponsory getDomainModel(Integer timeId) {
        LHResponsory theModel = new LHResponsory();
        theModel.setText(Utils.replaceByTime(getTexto(), timeId));
        theModel.setType(getTipo());
        theModel.setSource(getFuente());
        return theModel;
    }

}

File: app/src/main/java/org/deiverbum/app/data/entity/LHResponsoryEntity.kt
package org.deiverbum.app.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import org.deiverbum.app.model.LHResponsory
import org.deiverbum.app.util.Constants
import org.deiverbum.app.util.Utils

/**
 * Entidad para la tabla **`lh_responsory`** de la base de datos, que se ocupa de gestionar los responsorios de la Liturgia de las Horas.
 *
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Entity(tableName = Constants.LH_RESPONSORY)
data class LHResponsoryEntity (
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "responsoryID")
    val responsorioId:Int,

    @ColumnInfo(name = "text")
    val texto:String,

    @ColumnInfo(name = "source")
    val fuente:String,

    @ColumnInfo(name = "type")
    val tipo:Int
) {
    fun getDomainModel(timeId: Int?): LHResponsory {
        val theModel = LHResponsory()
        theModel.text = Utils.replaceByTime(texto, timeId!!)
        theModel.type = tipo
        theModel.source = fuente
        return theModel
    }
}
